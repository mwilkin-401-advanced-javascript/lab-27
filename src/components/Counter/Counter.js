import React from 'react';

class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 0,
    };
  }

  handleIncrementClick = (e) => {
    e.preventDefault();
    this.setState({ counter: this.state.counter + 1 });
  };

  handleDecrementClick = (e) => {
    e.preventDefault();
    this.setState({ counter: this.state.counter - 1 });
  };

  render() {
    // eslint-disable-next-line prefer-destructuring
    const counter = this.state.counter;
    let color;
    const colorIncrement = { color: 'green' };
    const colorDecrement = { color: 'red' };
    if (counter > 0) {
      color = colorIncrement;
    } else if (counter < 0) {
      color = colorDecrement;
    }

    return (
      <div className="counter">
        <h4 style={color} className="count">{this.state.counter}</h4>
        <a href="*" onClick={this.handleIncrementClick}>
          Increment
        </a>
        <a href="*" onClick={this.handleDecrementClick}>
          Decrement
        </a>
      </div>
    );
  }
}

export default Counter;
