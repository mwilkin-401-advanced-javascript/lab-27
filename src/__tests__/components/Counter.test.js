import React from 'react';
import Counter from '../../components/'