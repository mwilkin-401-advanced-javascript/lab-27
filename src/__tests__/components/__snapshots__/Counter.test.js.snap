// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> Snapshot Test renders correctly 1`] = `
<div
  className="counter"
>
  <h4
    className="count"
  >
    0
  </h4>
  <a
    href="*"
    onClick={[Function]}
  >
    Increment
  </a>
  <a
    href="*"
    onClick={[Function]}
  >
    Decrement
  </a>
</div>
`;
